src/main/java/fi/metatavu/metamind/bot/functions/AbstractFreemarkerMetaBotFunction.java

```diff
@@ -48,5 +69,74 @@ protected String getRenderedText(String templateName, Session session, IntentMat
     
     return result;
   }
+  
+  private List<FreemarkerMessageModel> getSessionMessages(fi.metatavu.metamind.persistence.models.Session metamindSession) {
+    List<Message> sessionMessages = messageController.listSessionMessages(metamindSession);
+    
+    List<FreemarkerMessageModel> result = new ArrayList<>(sessionMessages.size());
+    for (Message sessionMessage : sessionMessages) {
+      result.add(new FreemarkerMessageModel(getDate(sessionMessage.getCreated()), sessionMessage.getContent(), sessionMessage.getResponse()));
+    }
+    
+    return result;
+  }
+  
+  private Date getDate(OffsetDateTime offsetDateTime) {
+    return Date.from(offsetDateTime.toInstant());
+  }
+
+  public class FreemarkerModel {
+    
+    private Map<String, Object> sessionAttributes;
+    private List<FreemarkerMessageModel> sessionMessages;
+    public fi.metatavu.metamind.persistence.models.Session metamindSession;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make metamindSession a static final constant or non-public and provide accessors if needed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)